### ISSUE TITLE
Undefined variable: exitCode in vendor/symfony/console/Application.php on line 886

### ISSUE THREAD

**Issue by axzx** [symfony/symfony#25872]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 4.0.3

PHP 7.1.13-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Jan  5 2018 13:26:45) ( NTS )
env=prod

```
./bin/console c:cl

PHP Notice:  Undefined variable: exitCode in vendor/symfony/console/Application.php on line 886

In ConsoleTerminateEvent.php line 27:
                                                                                                                                                     
  Type error: Argument 4 passed to Symfony\Component\Console\Event\ConsoleTerminateEvent::__construct() must be of the type integer, null given, ca  
  lled in vendor/symfony/console/Application.php on line 886                                                                           
                                                                                                                                                     

cache:clear [--no-warmup] [--no-optional-warmers] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command>
```


## Reproduction

run:
```
composer create-project symfony/skeleton demo
cd demo
composer req mailer debug
```

edit `.env`
```
APP_ENV=prod
```

edit `config/packages/prod/monolog.yaml`
```
monolog:
    handlers:
        mail:
            type:         fingers_crossed
            action_level: critical
            handler:      deduplicated
        deduplicated:
            type:    deduplication
            handler: swift
        swift:
            type:       swift_mailer
            from_email: '%env(EMAIL_SENDER)%'
            to_email:   '%env(EMAIL_RECIPIENT)%'
            subject:    'Error %env(ROUTER_HOST)%'
            level:      debug
            formatter:  monolog.formatter.html
            content_type: text/html
```

run:
```
rm -rf var/cache/prod
./bin/console c:cl
```

**Comment by nicolas-grekas** [comment]
Can you share a reproducer or debug the situation? looking at the code in Application.php, I'm missing how this can happen.

**Comment by axzx** [comment]
`config/packages/prod/monolog.yaml`

```
monolog:
    handlers:
        mail:
            type:         fingers_crossed
            # 500 errors are logged at the critical level
            action_level: critical
            # to also log 400 level errors (but not 404's):
            # action_level: error
            # excluded_404s:
            #     - ^/
            handler:      deduplicated
        deduplicated:
            type:    deduplication
            handler: swift
        swift:
            type:       swift_mailer
            from_email: '%env(EMAIL_SENDER)%'
            to_email:   '%env(EMAIL_RECIPIENT)%'
```

But on production, in file `.env` the variable `EMAIL_RECIPIENT` does not exist. 
When I run the command `cache:clear` I get this message.

**Comment by nicolas-grekas** [comment]
Sorry, that's not enough to reproduce the issue on my side.

**Comment by axzx** [comment]
file `.env`:

```
APP_ENV=prod
# EMAIL_RECIPIENT=email@localhost
```


file `config/packages/prod/monolog.yaml`:

```
monolog:
    handlers:
        mail:
            type:         fingers_crossed
            action_level: critical
            handler:      deduplicated
        deduplicated:
            type:    deduplication
            handler: swift
        swift:
            type:       swift_mailer
            from_email: '%env(EMAIL_SENDER)%'
            to_email:   '%env(EMAIL_RECIPIENT)%'
            subject:    'Error %env(ROUTER_HOST)%'
            level:      debug
            formatter:  monolog.formatter.html
            content_type: text/html
```


run `rm -rf var/cache/prod; ./bin/console c:cl`

**Comment by theofidry** [comment]
Could you create a public repo on which can reproduce the issue? That would help to both narrow down the issue and avoid back and forth

**Comment by nicolas-grekas** [comment]
```sh
composer create-project "symfony/skeleton" demo
cd demo
composer req log
# edit .env as above
# edit config/packages/prod/monolog.yaml
rm -rf var/cache/prod; ./bin/console c:cl
```

=> The service "monolog.handler.swift.mail_message_factory" has a dependency o  
  n a non-existent service "mailer".

No notice.

**Comment by axzx** [comment]
@nicolas-grekas 
```
composer create-project symfony/skeleton demo
cd demo
composer req mailer debug
# edit .env as above
# edit config/packages/prod/monolog.yaml
rm -rf var/cache/prod; ./bin/console c:cl
```

**Comment by nicolas-grekas** [comment]
Works thanks, fixed in #25873
